pkgs/applications/graphics/cq-editor/default.nix

```
@@ -0,0 +1,57 @@
+{ lib
+, mkDerivationWith
+, python3Packages
+, fetchFromGitHub
+}:
+
+mkDerivationWith python3Packages.buildPythonApplication rec {
+  pname = "cq-editor";
+  version = "0.1RC1";
+
+  src = fetchFromGitHub {
+    owner = "CadQuery";
+    repo = "CQ-editor";
+    rev = version;
+    sha256 = "0iwcpnj15s64k16948sakvkn1lb4mqwrhmbxk3r03bczs0z33zax";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    cadquery
+    Logbook
+    pyqt5
+    pyparsing
+    pyqtgraph
+    spyder
+    pathpy
+    qtconsole
+    requests
+  ];
+
+  checkInputs = with python3Packages; [
+    pytest
+    pytest-xvfb
+    pytest-mock
+    pytestcov
+    pytest-repeat
+    pytest-qt
+  ];
+
+  postFixup = ''
+    wrapQtApp "$out/bin/cq-editor"
```

i would use wrapQtAppsHook

```suggestion
    nativeBuildInputs = [ wrapQtAppsHook];
```

---

It's been so long since I did this Qt stuff I'd forgotten some of it. Just reread https://nixos.org/nixpkgs/manual/#sec-language-qt and now I remember.

mkDerivationWith comes from libsForQt5.callpackage, and when you use mkDerivationWith you don't need to include wrapQtAppsHook.

The default wrapQtAppsHook ignores non-ELFs such as this python app, so I have to manually wrap it in postFixup.

---

that all sounds correct :)